### PR TITLE
Allow `growthbook auth login`  to accept its arguments as flags

### PR DESCRIPTION
**Pull request by AlexHartford**
The primary motivation behind this change is to allow running commands in a CI/automated environment. This also lets you do things like generate types for features with an npm command.

Fixes https://github.com/growthbook/growthbook-cli/issues/36

### CODE REVIEW

File: src/commands/auth/login.ts [hunk 1]
```diff
@@ -1,43 +1,71 @@
...
+    const {
+      args: {
+        apiKey,
+      },
+      flags: {
+        profile,
+        apiBaseUrl,
+      },
+    } = await this.parse(Login)
+    let profileUsed = profile || DEFAULT_GROWTHBOOK_PROFILE
```
Comment (tinahollygb) [inline src/commands/auth/login.ts:34]: This looks like if they don't pass in a profile as a flag, we will end up overwriting their default profile, and with this logic we will never prompt them for a profile below.

File: src/commands/auth/login.ts [hunk 2]
```diff
@@ -1,43 +1,71 @@
...
-    })
-    if (!profile) {
-      profile = DEFAULT_GROWTHBOOK_PROFILE
+    if (!profileUsed) {
+      profileUsed = await ux.prompt(`What is the name of this profile? You can leave this blank (default: ${DEFAULT_GROWTHBOOK_PROFILE})`, {
+        required: false,
+      })
+      if (!profileUsed) {
+        profileUsed = DEFAULT_GROWTHBOOK_PROFILE
+      }
```
Comment (tinahollygb) [inline src/commands/auth/login.ts:56]: This code will never run.

**Review by tinahollygb — CHANGES_REQUESTED** [review]
This adds regressions to the login command. Please see comments inline.

**Comment by AlexHartford** [comment]
> This adds regressions to the login command. Please see comments inline.



> This adds regressions to the login command. Please see comments inline.

Should be fixed now! @tinahollygb

File: src/commands/auth/login.ts (hunk 1 above)
Comment (AlexHartford) [reply src/commands/auth/login.ts]: Good catch, fixed!

File: src/commands/auth/login.ts (hunk 2 above)
Comment (AlexHartford) [reply src/commands/auth/login.ts]: Sure won't! Thanks for catching that.

**Review by tinahollygb — APPROVED** [review]

**Pull request merged**